src/OpenProtocolInterpreter/Header.cs

```diff
@@ -71,16 +73,17 @@ public Header()
 
         public override string ToString()
         {
-            string header = Length.ToString().PadLeft(4, '0');
-            header += Mid.ToString().PadLeft(4, '0');
-            header += (Revision > 0) ? Revision.ToString().PadLeft(3, '0') : "   ";
-            header += NoAckFlag ? "1" : " ";
-            header += (StationId != null) ? StationId.ToString().PadLeft(2, '0') : string.Empty.PadLeft(2, ' ');
-            header += (SpindleId != null) ? SpindleId.ToString().PadLeft(2, '0') : string.Empty.PadLeft(2, ' ');
-            header += (SequenceNumber > 0) ? SequenceNumber.ToString().PadLeft(2, '0') : string.Empty.PadLeft(2, ' ');
-            header += NumberOfMessages.ToString().PadLeft(1, ' ');
-            header += MessageNumber.ToString().PadLeft(1, ' ');
-            return header;
+            StringBuilder header = new StringBuilder();
+            header.Append(Length.ToString($"D{4}"));
+            header.Append(Mid.ToString($"D{4}"));
```

Comment:
Concatenating the number probably is a problem too, yet I'm not sure if compiler turns it into a constant at compilation time, but it's better to use `"D4"` than `$"D{4}"`